olexandr.vakulinskiy/HW2/Task1/tamagotchi.rb

```diff
@@ -0,0 +1,115 @@
+class Tamagotchi
+  attr_accessor :name, :health, :hunger, :thirst, :fatigue
+
+  def initialize(name)
+    @name = name
+    @health = 10
+    @hunger = 2
+    @thirst = 2
+    @fatigue = 0
+  end
+
+  def check_health
+    if @hunger.zero? || @thirst.zero?
+      @health += 1
+    elsif @hunger >= 7 || @thirst >= 7
+      @health -= 1
+    end
+
+    case @fatigue
+    when 10
+      @health = 0
+    when 7...10
+      @health -= 2
+    end
+  end
+
+  def feed
+    check_health
+    @hunger -= 1
+  end
+
+  def drink
+    check_health
+    @thirst -= 1
+  end
+
+  def play
+    check_health
+    @fatigue += 2
+    @hunger += 1
+    @thirst += 1
+  end
+
+  def sleep
+    check_health
+    @fatigue -= 1
+    @hunger += 2
+    @thirst += 2
+  end
+
+  def dead?(action_with_pet)
```

Comment:
@olexandr.vakulinskiy Note to you, predicate methods should only output boolean values (if you wish, you can correct the method and logic of the program now or not fixed it)